src/libtcod/bresenham.h

```diff
@@ -108,6 +112,26 @@ class TCODLIB_API BresenhamLine {
   */
   inline bool step(int& x, int& y) { return TCOD_line_step_mt(&x, &y, &data_); }
 
+  struct iterator : public std::iterator<std::input_iterator_tag, cell> {
+    iterator(value_type cur, TCOD_bresenham_data_t& data) : cur_(cur), data_(data) {}
+
+    inline reference operator*() { return cur_; }
+    inline iterator& operator++() {
+      TCOD_line_step_mt(&cur_.first, &cur_.second, &data_);
+      return *this;
+    }
+
+    inline bool operator==(const iterator& rhs) const { return cur_ == rhs.cur_; }
+    inline bool operator!=(const iterator& rhs) const { return cur_ != rhs.cur_; }
+
+   private:
+    value_type cur_;
+    TCOD_bresenham_data_t& data_;
+  };
+
+  iterator begin() { return iterator({data_.origx, data_.origy}, data_); }
+  iterator end() { return iterator({data_.destx, data_.desty}, data_); }
```

Comment:
Keep in mind `end` in C++ means the past-the-end iterator.  The current implementation here will clip the endpoint.